internal/storage/storage.go

```diff
@@ -0,0 +1,56 @@
+package storage
+
+import (
+	"time"
+
+	t "github.com/offen/docker-volume-backup/internal/types"
+)
+
+// Interface for defining functions which all storage providers support.
+type Backend interface {
+	Copy(file string) error
+	Prune(deadline time.Time, pruningPrefix string) error
+}
+
+// Generic type of storage. Everything here are common properties of all storage types.
+type StorageBackend struct {
```

Comment:
This is pretty academic, but I'll ask anyways: how much effort and duplication would removing this type introduce? If it's not that much I think it might be worth keeping the packages that contain the actual storage logic as isolated as possible. For example, `DoPrune` is a little odd now anyways as the most important part is passed in from the child package anyways.

Reply:
`StorageBackend` contains als fields which apply to all kinds of backends (5 fields as of now).
Not using such a generic storage backend and therefore not guaranteeing the existence of these mandatory fields, it would feel messy to me. Like it is now, you only have to add backend-specific fields to the actual backend's type struct (like AWS Bucket Name etc.), which is dependent on the backend technology.
About the DoPrune, I don't really a real opinion on that. It felt weird to do such a partial 'outsourcing' of a function to me, but if that's a common Go thing to do, I'm fine with it.

Reply:
> About the DoPrune, I don't really a real opinion on that. It felt weird to do such a partial 'outsourcing' of a function to me, but if that's a common Go thing to do, I'm fine with it.

It is super odd and definitely not a common Go thing to do, it was just the only way to achieve some deduplication of code in the current approach.

> Like it is now, you only have to add backend-specific fields to the actual backend's type struct (like AWS Bucket Name etc.), which is dependent on the backend technology.

The downside to this is that some hypothetical person that adds a new storage would have to learn about an additional concept on top of satisfying the one interface. But I also see your point. I'll move this around in the back of my head for a little, maybe some idea pops up.